[![DOI](https://zenodo.org/badge/DOI/10.5281/zenodo.7629625.svg)](https://doi.org/10.5281/zenodo.7629625)

[![Hits](https://hits.seeyoufarm.com/api/count/incr/badge.svg?url=https%3A%2F%2Fgithub.com%2Fatejaarsma%2FGR21_Greenland_Ice_Sheet_Microbial_Diversity_Data_Handling&count_bg=%2379C83D&title_bg=%23555555&icon=&icon_color=%23E7E7E7&title=hits&edge_flat=false)](https://hits.seeyoufarm.com)

# Scripts used for sequencing data handling 
## manuscript in prep by Ate H. Jaarsma
## ampvis.R
### creating heatmaps + alpha diversity measures
#### Andersen, Kasper S., Rasmus H. Kirkegaard, Soren M. Karst, and Mads Albertsen. 2018. “Ampvis2: An r Package to Analyse and Visualise 16s rRNA Amplicon Data.” bioRxiv. https://doi.org/10.1101/299537.

## metacoder.R
### creating heat trees
#### Foster, Zachary, Thomas Sharpton, and Niklaus Grünwald. 2017. “Metacoder: An r Package for Visualization and Manipulation of Community Taxonomic Diversity Data.” PLOS Computational Biology 13 (2): 1–15. https://doi.org/10.1371/journal.pcbi.1005404.

##  Qiime2_16S_18S.txt
### amplicon data analysis
#### Bolyen E, Rideout JR, Dillon MR et al. Reproducible, interactive, scalable and extensible microbiome data science using QIIME 2. Nat Biotechnol 2019 378 2019;37:852–7.
    
   
